site/docs/hosting/firebase.md

```diff
@@ -0,0 +1,182 @@
+   The `node` version should match your installed version of Node.js.
+   Otherwise, the project might not run.
+
+## Preparing Your Code
+
+You can use this short example bot as a starting point:
+
+```ts
+import * as functions from "firebase-functions";
+import { Bot, webhookCallback } from "grammy";
+
+const bot = new Bot(""); // <-- put your bot token here (https://t.me/BotFather)
+
+bot.command("start", (ctx) => ctx.reply("Welcome! Up and running."));
+bot.command("ping", (ctx) => ctx.reply(`Pong! ${new Date()}`));
+
+// During development, you can trigger your function at https://localhost/<firebase-projectname>/us-central1/helloworld
+export const helloWorld = functions.https.onRequest(webhookCallback(bot));
+```
+
+## Local Development
+
+During development, you can use the firebase emulator suite to run your code locally.
+This is way faster than deploying every change to Firebase.
+To install the emulators, run:
+
+```sh
+firebase init emulators
+```
+
+The functions emulator should be selected already.
+(If it isn't, navigate to it using the arrow keys, and select it using `space`.)
+For the questions about which port to use for each emulator, simply press `enter`.
+
+To start the emulators and run your code, use:
+
+```sh
+npm run serve
+```
+
+::: tip
+For some reason the standard configuration of the npm script does not start the TypeScript compiler in watch mode.
+Therefore, if you use TypeScript, you also have to run:
+
+```sh
+npm run build:watch
+```
+
+:::
+
+After the emulators start, you should find a line in the console output that looks like this:
+
+```sh
++  functions[us-central1-helloWorld]: http function initialized (http://127.0.0.1:5001/<firebase-projectname>/us-central1/helloWorld).
+```
+
+That is the local URL of your cloud function.
+However, your function is only available to the localhost on your computer.
+To actually test your bot, you need to expose your function to the internet so that the Telegram API can send updates to your bot.
+There are several services, such as [localtunnel](https://localtunnel.me) or [ngrok](https://ngrok.com), that can help you with that.
+In this example, we will use localtunnel, as it does not shut down after an hour like ngrok does.
+
+First, let's install localtunnel:
+
+```sh
+npm i -g localtunnel
+```
+
+After that, you can expose the port `5001` to the internet:
+
+```sh
+lt --port 5001
+```
+
+localtunnel should give you a unique URL, such as `https://modern-heads-sink-80-132-166-120.loca.lt`
+
+All that's left to do is to tell Telegram where to send the updates.
+You can do this by calling setWebhook.
+For example, open a new tab in your browser and visit this URL:
+
+```text:no-line-numbers
+https://api.telegram.org/bot<BOT_TOKEN>/setWebhook?url=<LT_URL>/<firebase-projectname>/us-central1/helloWorld
+```
+
+Replace `<BOT_TOKEN>` with your real bot token, and `<LT_URL>` with your own URL you received from localtunnel.
```

Comment:
```suggestion
https://api.telegram.org/bot<BOT_TOKEN>/setWebhook?url=<WEBHOOK_URL>/<firebase-projectname>/us-central1/helloWorld
```

Replace `<BOT_TOKEN>` with your real bot token, and `<WEBHOOK_URL>` with your own URL you got from localtunnel.
```